### PR TITLE
Fix realtype conversion in KafkaRecordCursor

### PR DESCRIPTION
**Pull request by elonazoulay**
Since RealType has javaType of long, float types should
be handled in getLong()

### CODE REVIEW

File: presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java [hunk 1]
```diff
@@ -166,8 +166,8 @@ public boolean isNull()
         @Override
```
Comment (kokosing) [inline presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java:166]: Can we have a test that would expose this bug, so the then we could the fix that actually fixes the problem (test)?

File: presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java [hunk 2]
```diff
@@ -166,8 +166,8 @@ public boolean isNull()
...
         public double getDouble()
         {
-            if (value instanceof Double || value instanceof Float) {
-                return ((Number) value).doubleValue();
```
Comment (findepi) [inline presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java:170]: Good catch. We should really avoid using `Number` interface.
It's not the first time it's causing correctness problems.

File: presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java [hunk 3]
```diff
@@ -187,6 +187,9 @@ public long getLong()
             if (value instanceof Long || value instanceof Integer) {
                 return ((Number) value).longValue();
             }
+            else if (value instanceof Float) {
+                return floatToIntBits((Float) value) & 0xFF_FF_FF_FFL;
```
Comment (findepi) [inline presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java:191]: the mask is not needed here, is it?

Comment (elonazoulay) [reply presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java:191]: Thanks:) It's not needed, will update.

Comment (elonazoulay) [reply presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java:191]: It's actually wrong - if value is negative, so definitely removing it.

File: presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java (hunk 1 above)
Comment (kokosing) [reply presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java]: ping

**Comment by mosabua** [comment]
:wave: @elonazoulay - this PR has become inactive. If you're still interested in working on it, please let us know, and we can try to get reviewers to help with that.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Pull request closed without merging**